code/ai/aicode.cpp

```diff
@@ -10006,7 +10007,7 @@ void guard_object_was_hit(object *guard_objp, object *hitter_objp)
 			aip->submode = SM_ATTACK;
 			aip->submode_start_time = Missiontime;
 			aip->active_goal = AI_ACTIVE_GOAL_DYNAMIC;
-		} else {
+		} else if (Objects[aip->target_objnum].type == OBJ_SHIP) {
```

Comment:
Do we need a new `else` clause? What should happen if neither `if`-condition applies?

Reply:
You're right.  We should put an unreachable in an else because it *should* be a previously defended ship.  But if it isn't, it would be the source of our bug, and so it brings us one step closer to finding the source.